File: src/Entity/Ordonnance.php
<?php

namespace App\Entity;

use App\Repository\OrdonnanceRepository;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\DBAL\Types\Types;
use DateTime;

#[ORM\Entity(repositoryClass: OrdonnanceRepository::class)]
class Ordonnance
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;
    #[ORM\Column]
      /**
     * @Assert\NotBlank(message="")
     * @Assert\Length(
     *      min = 2,
     *      minMessage="Entrer un titre au minimum de 2 caractères"
     * )
     */
    private ?string $nommedecin = null;
    #[ORM\Column]
       /**
     * @Assert\NotBlank(message="Le nom doit être non vide")
     * @Assert\Length(
     *      min = 2,
     *      minMessage="Entrer un titre au minimum de 2 caractères"
     * )
     */
    private ?string $nompatient = null;
    #[ORM\Column]
       /**
     * @Assert\NotBlank(message="La description doit être non vide")
     * @Assert\Length(
     *      min = 2,
     *      minMessage="Entrer un titre au minimum de 2 caractères"
     * )
     */
    private ?string $description = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\NotBlank(message: 'La date ne peut pas être vide')]
    #[Assert\Range(
        min: 'today',
        minMessage: 'La date ne peut pas être dans le passé',
        max: 'today',
        maxMessage: 'La date ne peut pas être dans le futur'
    )]
    private ?\DateTimeInterface $datedecreation = null;
    #[ORM\ManyToOne(inversedBy: 'pharmacies')]
    private ?Pharmacie $pharmacie = null;

    public function __construct()
    {
        $this->datedecreation = new DateTime();
    }
    public function getId(): ?int
    {
        return $this->id;
    }
    public function getNommedecin(): ?string
    {
        return $this->nommedecin;
    }

    public function setNommedecin(string $nommedecin): static
    {
        $this->nommedecin = $nommedecin;

        return $this;
    }
    public function getNompatient(): ?string
    {
        return $this->nompatient;
    }

    public function setNompatient(string $nompatient): static
    {
        $this->nompatient = $nompatient;

        return $this;
    }
    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }
    public function getDatedecreation(): ?\DateTimeInterface
    {
        return $this->datedecreation;
    }

    public function setDatedecreation(\DateTimeInterface $datedecreation): static
    {
        $this->datedecreation = $datedecreation;

        return $this;
    }
    
    
    public function getPharmacie(): ?Pharmacie
    {
        return $this->pharmacie;
    }
    
    public function setPharmacie(?Pharmacie $pharmacie): static
    {
        $this->pharmacie = $pharmacie;
    
        return $this;
    }
    
   
}

File: src/main/java/models/Ordonnance.java
package models;
import java.sql.Date;
import java.time.LocalDate;

public class Ordonnance {
        private int id;
        private String nommedecin;
        private String nompatient;
        private String description;
        private  int pharmacie_id;

    private String nompharmacie;


        // Constructeur
        public Ordonnance(){
        }
        public Ordonnance(String nommedecin, String nompatient, String description, int pharmacie_id,String nompharmacie) {
            this.nommedecin = nommedecin;
            this.nompatient = nompatient;
            this.description = description;
            this.pharmacie_id=pharmacie_id;
            this.nompharmacie=nompharmacie;

        }

        // Getters et setters
        public int getId(){return id;}
    public void setId(int id){
            this.id=id;
    }

        public String getNommedecin() {
            return nommedecin;
        }

        public void setNommedecin(String nommedecin) {
            this.nommedecin = nommedecin;
        }

        public String getNompatient() {
            return nompatient;
        }

        public void setNompatient(String nompatient) {
            this.nompatient = nompatient;
        }

        public String getDescription() {
            return description;
        }

        public void setDescription(String description) {
            this.description = description;
        }
    public int getPharmacieid() {
        return pharmacie_id;
    }

    public void setPharmacie_id(int pharmacie_id) {
        this.pharmacie_id = pharmacie_id;
    }
    public String getNompharmacie() {
        return nompharmacie;
    }

    public void setNompharmacie(String nompharmacie) {
        this.nompharmacie = nompharmacie;
    }

    @Override
    public String toString() {
        return "Ordonnance{"+
                "id=" + id +
                "nommedecin='" + nommedecin + '\'' +
                ", nompatient='" + nompatient + '\'' +
                ", description='" + description + '\'' +
                ",pharmacie id='"+pharmacie_id+
                ",nompharmacie '"+nompharmacie+

                '}';
    }


}
